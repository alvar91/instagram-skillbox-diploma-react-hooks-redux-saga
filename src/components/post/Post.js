import React from "react";
import PropTypes from "prop-types";
import { useSelector } from "react-redux";
import { usePostStyles } from "../../styles";
import UserCard from "../shared/UserCard";
import LikeButton from "../shared/LikeButton";
import { MoreIcon } from "../../icons";
import { Typography } from "@material-ui/core";
import OptionsDialog from "../shared/OptionsDialog";
import PostSkeleton from "./PostSkeleton";
import { getFormattedDate } from "../../utils/dateFormat";

const selectState = (state) => ({
  posts: state.posts.posts,
});

function Post({ id }) {
  const classes = usePostStyles();
  const [loading, setLoading] = React.useState(true);
  const [showOptionsDialog, setOptionsDialog] = React.useState(false);

  const state = useSelector(selectState);

  const {
    urls: { small: media },
    likes,
    user,
    updated_at,
    liked_by_user,
    alt_description = "Empty description",
    links: { html: linkToPost },
  } = state.posts.find((post) => post.id === id);

  setTimeout(() => setLoading(false), 1000);
  if (loading) return <PostSkeleton />;

  return (
    <div className={classes.postContainer}>
      <article className={classes.article}>
        {/* Post Header */}
        <div className={classes.postHeader}>
          <UserCard user={user} avatarSize={32} />
          <MoreIcon
            className={classes.moreIcon}
            onClick={() => setOptionsDialog(true)}
          />
        </div>
        {/* Post Image */}
        <div className={classes.postImage}>
          <img src={media} alt="Post media" className={classes.image} />
        </div>
        {/* Post Buttons */}
        <div className={classes.postButtonsWrapper}>
          <div className={classes.postButtons}>
            <LikeButton id={id} liked={liked_by_user} />
          </div>
          {likes > 0 && (
            <Typography className={classes.likes} variant="subtitle2">
              <span>{likes === 1 ? "1 like" : `${likes} likes`}</span>
            </Typography>
          )}
          <div className={classes.postCaptionContainer}>
            <Typography
              variant="body2"
              component="span"
              className={classes.postCaption}
              dangerouslySetInnerHTML={{ __html: alt_description }}
            />
          </div>
          <Typography color="textSecondary" className={classes.datePosted}>
            {getFormattedDate(updated_at)}
          </Typography>
        </div>
      </article>
      {showOptionsDialog && (
        <OptionsDialog
          linkToPost={linkToPost}
          onClose={() => setOptionsDialog(false)}
        />
      )}
    </div>
  );
}

Post.propTypes = {
  id: PropTypes.string.isRequired,
};

export default React.memo(Post);
