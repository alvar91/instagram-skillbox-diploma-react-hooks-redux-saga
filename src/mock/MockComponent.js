import React from "react";

function MockComponent() {
  return <h2>Mock Component</h2>;
}

export default MockComponent;
