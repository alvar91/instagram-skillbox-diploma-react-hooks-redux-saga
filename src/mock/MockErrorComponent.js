import React from "react";

function MockErrorComponent() {
  React.useEffect(() => {
    throw new Error("Mock error");
  }, []);

  return <h2>Mock error</h2>;
}

export default MockErrorComponent;
