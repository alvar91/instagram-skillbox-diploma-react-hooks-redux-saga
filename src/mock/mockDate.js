export const mockDate = "2021-02-11T17:25:19-05:00";
export const correctMockDate = "Fri, Feb 12, 2021";
