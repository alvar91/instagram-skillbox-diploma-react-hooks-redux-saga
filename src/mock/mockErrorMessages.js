export const mockAuthErrorMessage =
  "Failed to get access token. Please try to log in again";
export const mockErrorMeta = "meta error";
