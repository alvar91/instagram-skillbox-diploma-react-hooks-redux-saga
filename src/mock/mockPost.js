export const mockPost = {
  id: "1gLdTsX3_70",
  created_at: "2020-07-01T18:30:14-04:00",
  updated_at: "2021-02-11T23:12:42-05:00",
  promoted_at: null,
  width: 6016,
  height: 4016,
  color: "#260c0c",
  blur_hash: "LHEf7wbbM{w[_NM{-:xZRiVs%gNG",
  description: null,
  alt_description: "laptop on brown sofa",
  urls: {
    raw:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1",
    full:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
    regular:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
    small:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400",
    thumb:
      "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=200",
  },
  links: {
    self: "https://api.unsplash.com/photos/1gLdTsX3_70",
    html: "https://unsplash.com/photos/1gLdTsX3_70",
    download: "https://unsplash.com/photos/1gLdTsX3_70/download",
    download_location: "https://api.unsplash.com/photos/1gLdTsX3_70/download",
  },
  categories: [],
  likes: 738,
  liked_by_user: false,
  current_user_collections: [],
  sponsorship: {
    impression_urls: [
      "https://secure.insightexpressai.com/adServer/adServerESI.aspx?script=false&bannerID=8281535&rnd=[timestamp]&gdpr=&gdpr_consent=&redir=https://secure.insightexpressai.com/adserver/1pixel.gif",
    ],
    tagline: "Designed to be the Best",
    tagline_url: "http://www.dell.com/xps",
    sponsor: {
      id: "2DC3GyeqWjI",
      updated_at: "2021-02-12T14:33:51-05:00",
      username: "xps",
      name: "XPS",
      first_name: "XPS",
      last_name: null,
      twitter_username: "Dell",
      portfolio_url: "http://www.dell.com/xps",
      bio:
        "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
      location: null,
      links: {
        self: "https://api.unsplash.com/users/xps",
        html: "https://unsplash.com/@xps",
        photos: "https://api.unsplash.com/users/xps/photos",
        likes: "https://api.unsplash.com/users/xps/likes",
        portfolio: "https://api.unsplash.com/users/xps/portfolio",
        following: "https://api.unsplash.com/users/xps/following",
        followers: "https://api.unsplash.com/users/xps/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "dell",
      total_collections: 0,
      total_likes: 0,
      total_photos: 22,
      accepted_tos: true,
    },
  },
  user: {
    id: "2DC3GyeqWjI",
    updated_at: "2021-02-12T14:33:51-05:00",
    username: "xps",
    name: "XPS",
    first_name: "XPS",
    last_name: null,
    twitter_username: "Dell",
    portfolio_url: "http://www.dell.com/xps",
    bio:
      "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
    location: null,
    links: {
      self: "https://api.unsplash.com/users/xps",
      html: "https://unsplash.com/@xps",
      photos: "https://api.unsplash.com/users/xps/photos",
      likes: "https://api.unsplash.com/users/xps/likes",
      portfolio: "https://api.unsplash.com/users/xps/portfolio",
      following: "https://api.unsplash.com/users/xps/following",
      followers: "https://api.unsplash.com/users/xps/followers",
    },
    profile_image: {
      small:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
      medium:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
      large:
        "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
    },
    instagram_username: "dell",
    total_collections: 0,
    total_likes: 0,
    total_photos: 22,
    accepted_tos: true,
  },
};
