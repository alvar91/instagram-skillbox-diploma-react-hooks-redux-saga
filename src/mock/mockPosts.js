export const mockPosts = [
  {
    id: "1gLdTsX3_70",
    created_at: "2020-07-01T18:30:14-04:00",
    updated_at: "2021-02-11T23:12:42-05:00",
    promoted_at: null,
    width: 6016,
    height: 4016,
    color: "#260c0c",
    blur_hash: "LHEf7wbbM{w[_NM{-:xZRiVs%gNG",
    description: null,
    alt_description: "laptop on brown sofa",
    urls: {
      raw:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1",
      full:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1593642532454-e138e28a63f4?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=200",
    },
    links: {
      self: "https://api.unsplash.com/photos/1gLdTsX3_70",
      html: "https://unsplash.com/photos/1gLdTsX3_70",
      download: "https://unsplash.com/photos/1gLdTsX3_70/download",
      download_location: "https://api.unsplash.com/photos/1gLdTsX3_70/download",
    },
    categories: [],
    likes: 738,
    liked_by_user: false,
    current_user_collections: [],
    sponsorship: {
      impression_urls: [
        "https://secure.insightexpressai.com/adServer/adServerESI.aspx?script=false&bannerID=8281535&rnd=[timestamp]&gdpr=&gdpr_consent=&redir=https://secure.insightexpressai.com/adserver/1pixel.gif",
      ],
      tagline: "Designed to be the Best",
      tagline_url: "http://www.dell.com/xps",
      sponsor: {
        id: "2DC3GyeqWjI",
        updated_at: "2021-02-12T14:33:51-05:00",
        username: "xps",
        name: "XPS",
        first_name: "XPS",
        last_name: null,
        twitter_username: "Dell",
        portfolio_url: "http://www.dell.com/xps",
        bio:
          "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
        location: null,
        links: {
          self: "https://api.unsplash.com/users/xps",
          html: "https://unsplash.com/@xps",
          photos: "https://api.unsplash.com/users/xps/photos",
          likes: "https://api.unsplash.com/users/xps/likes",
          portfolio: "https://api.unsplash.com/users/xps/portfolio",
          following: "https://api.unsplash.com/users/xps/following",
          followers: "https://api.unsplash.com/users/xps/followers",
        },
        profile_image: {
          small:
            "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
          medium:
            "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
          large:
            "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
        },
        instagram_username: "dell",
        total_collections: 0,
        total_likes: 0,
        total_photos: 22,
        accepted_tos: true,
      },
    },
    user: {
      id: "2DC3GyeqWjI",
      updated_at: "2021-02-12T14:33:51-05:00",
      username: "xps",
      name: "XPS",
      first_name: "XPS",
      last_name: null,
      twitter_username: "Dell",
      portfolio_url: "http://www.dell.com/xps",
      bio:
        "Designed to be the best, with cutting edge technologies, exceptional build quality, unique materials and powerful features.",
      location: null,
      links: {
        self: "https://api.unsplash.com/users/xps",
        html: "https://unsplash.com/@xps",
        photos: "https://api.unsplash.com/users/xps/photos",
        likes: "https://api.unsplash.com/users/xps/likes",
        portfolio: "https://api.unsplash.com/users/xps/portfolio",
        following: "https://api.unsplash.com/users/xps/following",
        followers: "https://api.unsplash.com/users/xps/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1600096866391-b09a1a53451aimage?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "dell",
      total_collections: 0,
      total_likes: 0,
      total_photos: 22,
      accepted_tos: true,
    },
  },
  {
    id: "lqFM73nWnf8",
    created_at: "2021-02-12T05:41:16-05:00",
    updated_at: "2021-02-12T06:17:12-05:00",
    promoted_at: "2021-02-12T06:04:16-05:00",
    width: 5760,
    height: 6121,
    color: "#595959",
    blur_hash: "LD7d%rt7ofof00Rjayay-;j[j[ay",
    description: "Architecture abstract",
    alt_description: "gray scale photo of palm tree",
    urls: {
      raw:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1",
      full:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1613126406251-6ea13dcd62ed?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHw5fHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=200",
    },
    links: {
      self: "https://api.unsplash.com/photos/lqFM73nWnf8",
      html: "https://unsplash.com/photos/lqFM73nWnf8",
      download: "https://unsplash.com/photos/lqFM73nWnf8/download",
      download_location: "https://api.unsplash.com/photos/lqFM73nWnf8/download",
    },
    categories: [],
    likes: 31,
    liked_by_user: false,
    current_user_collections: [],
    sponsorship: null,
    user: {
      id: "r01XoGoEPT8",
      updated_at: "2021-02-12T14:13:55-05:00",
      username: "ripato",
      name: "Ricardo Gomez Angel",
      first_name: "Ricardo",
      last_name: "Gomez Angel",
      twitter_username: null,
      portfolio_url: "https://www.flickr.com/photos/rigoan/",
      bio:
        "I'm architect, marketing specialist and photographer. I`m working since 2 years as freelancer photographer for different Swiss companies",
      location: "Switzerland",
      links: {
        self: "https://api.unsplash.com/users/ripato",
        html: "https://unsplash.com/@ripato",
        photos: "https://api.unsplash.com/users/ripato/photos",
        likes: "https://api.unsplash.com/users/ripato/likes",
        portfolio: "https://api.unsplash.com/users/ripato/portfolio",
        following: "https://api.unsplash.com/users/ripato/following",
        followers: "https://api.unsplash.com/users/ripato/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1492187901140-48ac7eff801c?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1492187901140-48ac7eff801c?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1492187901140-48ac7eff801c?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "ricardogo03",
      total_collections: 0,
      total_likes: 14717,
      total_photos: 1159,
      accepted_tos: true,
    },
  },
  {
    id: "W3tpb2wGRfo",
    created_at: "2021-02-11T10:13:29-05:00",
    updated_at: "2021-02-12T06:17:12-05:00",
    promoted_at: "2021-02-12T05:12:04-05:00",
    width: 2277,
    height: 2664,
    color: "#73a6a6",
    blur_hash: "LJI}|o0fCm^+BWozE1WVK6xa-VNG",
    description: null,
    alt_description: "white concrete building under blue sky during daytime",
    urls: {
      raw:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1",
      full:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?crop=entropy&cs=srgb&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1&q=85",
      regular:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1&q=80&w=1080",
      small:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1&q=80&w=400",
      thumb:
        "https://images.unsplash.com/photo-1613056370848-9edd9d6e15b5?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwxOTY5OTJ8MHwxfGFsbHwxMHx8fHx8fDJ8&ixlib=rb-1.2.1&q=80&w=200",
    },
    links: {
      self: "https://api.unsplash.com/photos/W3tpb2wGRfo",
      html: "https://unsplash.com/photos/W3tpb2wGRfo",
      download: "https://unsplash.com/photos/W3tpb2wGRfo/download",
      download_location: "https://api.unsplash.com/photos/W3tpb2wGRfo/download",
    },
    categories: [],
    likes: 52,
    liked_by_user: false,
    current_user_collections: [],
    sponsorship: null,
    user: {
      id: "E37oh-Tr6bo",
      updated_at: "2021-02-12T14:13:54-05:00",
      username: "bennmcguinness",
      name: "Benn McGuinness",
      first_name: "Benn",
      last_name: "McGuinness",
      twitter_username: null,
      portfolio_url: null,
      bio: null,
      location: "London ",
      links: {
        self: "https://api.unsplash.com/users/bennmcguinness",
        html: "https://unsplash.com/@bennmcguinness",
        photos: "https://api.unsplash.com/users/bennmcguinness/photos",
        likes: "https://api.unsplash.com/users/bennmcguinness/likes",
        portfolio: "https://api.unsplash.com/users/bennmcguinness/portfolio",
        following: "https://api.unsplash.com/users/bennmcguinness/following",
        followers: "https://api.unsplash.com/users/bennmcguinness/followers",
      },
      profile_image: {
        small:
          "https://images.unsplash.com/profile-1611603138308-604e48efa757image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
        medium:
          "https://images.unsplash.com/profile-1611603138308-604e48efa757image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=64&w=64",
        large:
          "https://images.unsplash.com/profile-1611603138308-604e48efa757image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128",
      },
      instagram_username: "benn.mcguinness",
      total_collections: 1,
      total_likes: 1583,
      total_photos: 448,
      accepted_tos: true,
    },
  },
];
