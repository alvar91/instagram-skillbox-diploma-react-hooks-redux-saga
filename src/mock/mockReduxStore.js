// Core
import { createStore, applyMiddleware } from "redux";

// Instruments
import { rootReducer } from "../redux/rootReducer";
import {
  composeEnhancers,
  middleware,
  sagaMiddleware,
} from "../redux/middleware";
import { rootSaga } from "../redux/rootSaga";

// Mock state
import { mockFactoryStore } from "./mockFactoryStore";

export const mockReduxStore = (storeOptionsObj) => {
  const initialState = mockFactoryStore(storeOptionsObj);
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(...middleware))
  );

  sagaMiddleware.run(rootSaga);
  return store;
};
