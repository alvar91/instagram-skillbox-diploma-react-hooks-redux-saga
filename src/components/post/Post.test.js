// Core
import React from "react";
import { configure, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Post from "./Post";
import { mockPosts } from "../../mock/mockPosts";
import { mockPost } from "../../mock/mockPost";
import { PostSkeleton } from "./PostSkeleton";

// Instruments
import { Provider } from "react-redux";
import { mockReduxStore } from "../../mock/mockReduxStore";

// Utils
// import { delay } from "../../utils/delay";

configure({ adapter: new Adapter() });

/**
 * Setup function for Post component
 * @param {object} storeOptionsObj contains options for creating the mock store
 * @returns {ReactRenderer}
 */
const setupTestRenderer = (storeOptionsObj) => {
  const mockStore = mockReduxStore(storeOptionsObj);
  return mount(
    <Provider store={mockStore}>
      <Post id={mockPost.id} />
    </Provider>
  );
};

describe("Post", () => {
  describe("Post with mock post state and skeleton", () => {
    let testRenderer;
    let component;
    const props = { id: mockPost.id };

    let setOptionsDialog = jest.fn();

    beforeAll(() => {
      setOptionsDialog.mockClear();
      React.useState = jest.fn(() => [true, setOptionsDialog]);

      const storeOptionsObj = {
        postsObj: {
          posts: mockPosts,
          currentPage: 1,
          isFetchingPost: false,
          hasErrorPosts: false,
          errorMessagePosts: "",
        },
        usersObj: {
          users: [],
        },
        authObj: {
          isFetchingAuth: false,
          isLoggedIn: false,
          profile: [],
          hasErrorAuth: false,
          errorMessageAuth: "",
        },
      };
      testRenderer = setupTestRenderer(storeOptionsObj);
      component = testRenderer.find(Post);
    });

    it("Post should correctly renders markup", () => {
      expect(shallowToJson(testRenderer)).toMatchSnapshot();
    });

    it("Post should contain PostSkeleton", () => {
      expect(component).toHaveLength(1);
      expect(component.find(PostSkeleton)).toHaveLength(1);
    });

    it("Post should contain correct props", () => {
      expect(component).toHaveLength(1);
      expect(component.props()).toMatchObject(props);
    });

    // it("Post should`t contain PostSkeleton", async() => {
    //   await delay(2000)
    //   testRenderer.update();
    //   component = testRenderer.find(Post);
    //   expect(component.find(PostSkeleton)).toHaveLength(0);
    // });

    // it("img should contain correct props", () => {
    //   expect(component).toHaveLength(1);
    //   expect(component.find("img")).toHaveLength(1);
    // });
  });
});
