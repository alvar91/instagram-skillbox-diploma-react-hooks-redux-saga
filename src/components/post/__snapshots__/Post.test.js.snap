// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Post with mock post state and skeleton Post should correctly renders markup 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Memo(Post)
    id="1gLdTsX3_70"
  >
    <PostSkeleton>
      <div
        className="makeStyles-container-27"
        style={
          Object {
            "gridTemplateColumns": false,
          }
        }
      >
        <div
          className="makeStyles-mediaSkeleton-33"
        />
        <div>
          <div
            className="makeStyles-headerSkeleton-28"
          >
            <div
              className="makeStyles-avatarSkeleton-29"
            />
            <div
              className="makeStyles-headerTextSkeleton-30"
            >
              <div
                className="makeStyles-primaryTextSkeleton-31"
              />
              <div
                className="makeStyles-secondaryTextSkeleton-32"
              />
            </div>
          </div>
        </div>
      </div>
    </PostSkeleton>
  </Memo(Post)>
</Provider>
`;
